File: js/171_ExcelSheetColumnNumber.js
/**
 * @param {string} s
 * @return {number}
 */
var titleToNumber = function(s) {
    if( s.length === 0 ) return 0;
    var len = s.length;
    var codeA = 'A'.charCodeAt();
    var ret = 0;
    for( var i = 0; i < len; ++i ){
        ret = ret * 26 + (s[i].charCodeAt() - codeA + 1);
    }
    return ret;
};

File: cpp/171_ExcelSheetColumnNumber.cpp
class Solution {
public:
    int titleToNumber(string s) {
        if( s.empty() ) return 0;
        int len = s.length();
        int ret = 0;
        for( int i = 0; i < len; ++i ){
            ret = ret * 26 + (s[i] - 'A' + 1);
        }
        return ret;
    }
};